plugins/terraform/stepper.go

```diff
@@ -356,47 +356,87 @@ func GetBackendConfig(exec config.StepExecution, backendParser TFBackendParser)
 	// if user has decided to overwrite state file convention in backend.tf, support this override
 	if declaredBackend.Key != "" {
 		// grab statefile name (base)
-		b["key"] = interpolateString(exec, declaredBackend.Key)
+		interpolatedKey := interpolateString(exec, declaredBackend.Key)
+		b["key"] = interpolatedKey
 	}
 
-	if declaredBackend.S3RoleArn != "" {
-		b["role_arn"] = interpolateString(exec, declaredBackend.S3RoleArn)
+	if declaredBackend.AssumeRole.RoleArn != "" {
+		roleArn := declaredBackend.AssumeRole.RoleArn
+		exec.Logger.Tracef("Declared S3RoleArn: %s", roleArn)
+		interpolatedRoelArn := interpolateString(exec, declaredBackend.AssumeRole.RoleArn)
 
-		exec.Logger.Debugf("Declared S3RoleArn: %s", b["role_arn"])
+		// only track override config if interpolated is different from what user declared
+		if interpolatedRoelArn != roleArn {
+			roleArn = interpolatedRoelArn
```

Comment:
Looks like a consistently typo named variable.  `s/interpolatedRoelArn/interpolatedRoleArn/g`

Reply:
thanks for catching! 👍